CRM/Volunteer/Form/VolunteerSignUp.php

```diff
@@ -285,6 +285,7 @@ function postProcess() {
   function buildCustom() {
     $contactID = CRM_Utils_Array::value('userID', $_SESSION['CiviCRM']);
     $profiles = array();
+    $fieldList = array(); // master field list
```

Comment:
This and the code that uses it below is the fix for the bug reported in the comments of VOL-58 causing improprer multiple-profile display.